packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -4071,6 +4088,7 @@ class _RenderEditableCustomPaint extends RenderBox {
     assert(parent != null);
     final RenderEditablePainter? painter = this.painter;
     if (painter != null && parent != null) {
+      parent._computeTextMetricsIfNeeded();
```

Comment:
Any potential performance impact from this and/or the removal of the conditional in _layoutText?

Reply:
- removal the conditional:

I think `TextPainter.layout` is already doing a good enough job caching the layout results. Most of the time when ` _textLayoutLastMaxWidth == maxWidth && _textLayoutLastMinWidth == minWidth` evaluates to true in `_layoutText` , ` !_needsLayout && minWidth == _lastMinWidth && maxWidth == _lastMaxWidth` in `TextPainter.layout` would return true anyways (so it's just a few additional instructions).
in cases where the former evaluates to true but the latter evaluates to false, the cache is either already invalidated ( `_needsLayout` is true, so accessing the text layout will crash), or the cache was computed using a different set of constraints and should be invalidated.

- adding `_computeTextMetricsIfNeeded` here: 

Yeah I think the multibox adapter should prevent this render object from painting since it didn't layout. But since its constraints are not changed (unless `computeDryLayout` or one of the compute intrinsics methods were called after the last `performLayout` call), the cache is likely going to hit if it's not invalidated.

Reply:
There's an edge case where `_textLayoutLastMaxWidth == maxWidth && _textLayoutLastMinWidth == minWidth` is not sufficient. See the test added in `editable_test.dart`.